app/gui2/src/components/GraphEditor/collapsing.ts

```diff
@@ -239,11 +235,13 @@ if (import.meta.vitest) {
   const { test, expect } = import.meta.vitest
 
   function setupGraphDb(code: string, graphDb: GraphDb) {
-    const ast = Ast.parseTransitional(code, new IdMap())
-    const expressions = Array.from(ast.statements())
+    const { root, toRaw, getSpan } = Ast.parseExtended(code)
+    const expressions = Array.from(root.statements())
     const func = expressions[0]
     assert(func instanceof Ast.Function)
-    graphDb.readFunctionAst(func, () => undefined)
+    const rawFunc = toRaw.get(func.id)
+    assert(rawFunc?.type === RawAst.Tree.Type.Function)
+    graphDb.readFunctionAst(func, rawFunc, code, (_id) => undefined, getSpan)
```

Comment:
The metadata getter won't be necessary once metadata is stored attached to the nodes.

The rest of it is for alias analysis. AA requires all the extra data `parseExtended` produces because it uses raw ASTs and relies on span information (which is incompatible with fully incremental updates). At some point we should refactor it to use the new APIs, and also make it incremental by tracking scopes in a way that doesn't require computing spans.